common/compositor/va/varenderer.cpp

```diff
@@ -150,23 +150,40 @@ bool VARenderer::SetVAProcFilterColorValue(HWCColorControl mode,
   }
 }
 
-bool VARenderer::GetVAProcDeinterlaceFlagFromVideo(HWCDeinterlaceFlag flag) {
+bool VARenderer::GetVAProcDeinterlaceFlagFromVideo(
+    const HWCDeinterlaceFlag flag, const OverlayLayer* layer) {
+  if (layer == NULL) {
+    ETRACE("Invalid layer");
+  }
+
   if (flag != HWCDeinterlaceFlag::kDeinterlaceFlagAuto) {
     return false;
   } else {
-    // TODO:Need video buffer meta data to judge if the frame really need
-    // Deinterlace.
+    // Inspect video buffer metadata in current layer to determine if
+    // this frame needs to be deinterlaced.
+    HWCDeinterlaceControl mode = layer->GetBuffer()->GetInterlaceMode();
+    DUMPTRACE("Interlace mode = (%d)", mode);
+    if (mode != HWCDeinterlaceControl::kDeinterlaceNone) {
```

Comment:
Do not confuse IsInterlaced with Deinterlace controller. It is different. IsInterlaced is about the buffer itself, Deinterlace controller is tha Application intend

Reply:
Same comment as above: "The initial intent was to read the interlace mode from the active layer but it looks like it's not necessary."